libmscore/textbase.cpp

```diff
@@ -328,6 +328,52 @@ bool TextCursor::movePosition(QTextCursor::MoveOperation op, QTextCursor::MoveMo
       return true;
       }
 
+
+
+//---------------------------------------------------------
+//   doubleClickSelect
+//---------------------------------------------------------
+
+void TextCursor::doubleClickSelect()
+      {
+      clearSelection();
+
+      if (!currentCharacter().isSpace()) {
+            //handle double-clicking inside a word
+            int startPosition = _column;
+
+            while (_column > 0 && !currentCharacter().isSpace())
+                  --_column;
+
+            if (currentCharacter().isSpace())
+                  ++_column;
+
+            _selectColumn = _column;
+
+            _column = startPosition;
+            while (_column  < curLine().columns() && !currentCharacter().isSpace())
+                  ++_column;
+            }
+      else {
+            //handle double-clicking between words
+            int startPosition = _column;
+
+            while (_column > 0 && currentCharacter().isSpace())
+                  --_column;
+
+            if (!currentCharacter().isSpace())
+                  ++_column;
+
+            _selectColumn = _column;
+
+            _column = startPosition;
+            while (_column  < curLine().columns() && currentCharacter().isSpace())
+                  ++_column;
```

Comment:
The selection code is good but these two branches of `if` statement are absolutely equal except the fact that one is searches for space characters and other — for non-space characters. Can they be perhaps merged to avoid the code duplication? As a simple option, we could make some boolean flag, say,
`const bool selectSpaces = currentCharacher().isSpace();`
and use it in conditions in this code, for example
```
while (_column > 0 && (currentCharacter().isSpace() == selectSpaces))
      --_column;
// etc.
```
Then we won't need to write this code twice.

Reply:
OK. I've done that. I think it's a wee bit harder to follow the code now, but it does reduce duplication.